Tests/kaas/k8s-version-policy/k8s_version_policy.py

```diff
@@ -147,127 +147,88 @@ def initialize_config(config):
+@dataclass(order=True)
+class K8sVersion:
+    major: int
+    minor: int
+    patch: int = 0
 
-    Attributes:
-        major (int): Major version of the k8s version
-        minor (int): Minor version of the k8s version
-        patch (int): Patch version of the k8s version
-        date (datetime): release date of the k8s version
-    """
-    def __init__(self, major=0, minor=0, patch=0):
-        self.major = major
-        self.minor = minor
-        self.patch = patch
-
-        self.date = None
-
-    def __eq__(self, other):
-        if not isinstance(other, K8sVersionInfo):
-            raise TypeError
-        return self.major == other.major and self.minor == other.minor and self.patch == other.patch
-
-    def __gt__(self, other):
-        if not isinstance(other, K8sVersionInfo):
-            raise TypeError
-        patchcomp = self.minor == other.minor and self.patch > other.patch
-        return self.major > other.major or (self.major == other.major and (self.minor > other.minor or patchcomp))
-
-    def __ge__(self, other):
-        if not isinstance(other, K8sVersionInfo):
-            raise TypeError
-        patchcomp = self.minor == other.minor and self.patch >= other.patch
-        return self.major > other.major or (self.major == other.major and (self.minor > other.minor or patchcomp))
-
-    def __lt__(self, other):
-        if not isinstance(other, K8sVersionInfo):
-            raise TypeError
-        patchcomp = self.minor == other.minor and self.patch < other.patch
-        return self.major < other.major or (self.major == other.major and (self.minor < other.minor or patchcomp))
-
-    def __le__(self, other):
-        if not isinstance(other, K8sVersionInfo):
-            raise TypeError
-        patchcomp = self.minor == other.minor and self.patch <= other.patch
-        return self.major < other.major or (self.major == other.major and (self.minor < other.minor or patchcomp))
-
-    @classmethod
-    def extract_version(cls, string, separator=".", strip=None):
-        if strip is None:
-            strip = ["v"]
-        for s in strip:
-            string = string.strip(s)
-        components = string.strip().split(separator)
-        return cls(int(components[0]), int(components[1]), int(components[2]))
-
-    def check_for_version(self, major=None, minor=None, patch=None):
-        """Check if a version or part of the version is equal to the given version numbers"""
-        return (major is None or self.major == major) and \
-               (minor is None or self.minor == minor) and \
-               (patch is None or self.patch == patch)
+    def branch(self):
+        """Return a tuple of only the major and minor version."""
+        return (self.major, self.minor)
 
     def __str__(self):
         return f"{self.major}.{self.minor}.{self.patch}"
 
 
-class CVEVersionInfo:
-    """Class that contains a CVE version info.
+def parse_version(version_str: str) -> K8sVersion:
```

Comment:
I like this very much. I'm wondering though why you opted against a classmethod.

Reply:
Personal preference, there is no *real* technical reason to do so :innocent: I prefer it because it is shorter to write without the class prefix and you would either need to drop the type hints or add `from __future__ import annotations` to make the interpreter happy about the reference of `K8sVersion` return type before the class definition is finished.